### PR TITLE
init connection #14

### LINKED ISSUE #14: Initialize BitShares connection
Initialize bitshares connection

Use vuex-bitshares "Connection" module for it. Create `init` action in  the`app` vuex module, that should dispatch connection action.

### PR DESCRIPTION
**Pull request by ololokiras**
close #14 create action for init connection

### CODE REVIEW

File: src/store/modules/app.js [hunk 1]
```diff
@@ -6,11 +8,16 @@ const getters = {
 }
 
 const mutations = {
-
+  [APP_INIT] () {
```
Comment (roma219) [inline src/store/modules/app.js]: no need for mutation if it doesn't modify state

File: src/main.js [hunk 2]
```diff
@@ -10,3 +10,5 @@ new Vue({
   store,
   render: h => h(App)
 }).$mount('#app')
+
+store.dispatch('connection/initConnection')
```
Comment (roma219) [inline src/main.js]: It's better to do it before app initialization (before "new Vue...")

**Review by roma219 — CHANGES_REQUESTED** [review]
Good, but couple smallfixes required

File: src/store/modules/app.js [hunk 3]
```diff
@@ -1,3 +1,5 @@
+const APP_INIT = 'APP_INIT'
```
Comment (roma219) [inline src/store/modules/app.js]: no need for mutation that does nothing :0

File: src/store/modules/app.js [hunk 4]
```diff
@@ -6,11 +8,14 @@ const getters = {
...
 const mutations = {
-
 }
 
 const actions = {
-
+  init (store) {
+    const { commit } = store
+    store.dispatch('connection/initConnection', null, { root: true })
+    commit(APP_INIT)
```
Comment (roma219) [inline src/store/modules/app.js]: and that too

**Review by roma219 — CHANGES_REQUESTED** [review]
Please remove that mutation

**Review by roma219 — APPROVED** [review]
👍

File: src/main.js [hunk 5]
```diff
@@ -5,6 +5,8 @@ import store from './store/'
 
 Vue.config.productionTip = false
 
+store.dispatch('connection/initConnection')
```
Comment (roma219) [inline src/main.js]: dispatch - 'app/init', in future this action is gonna have the just connection establishment, but we want a clean way to do it all at once from main.js

**Review by roma219 — CHANGES_REQUESTED** [review]

**Review by roma219 — APPROVED** [review]
👍

**Comment by roma219** [comment]
@dls-cipher @hisinbox merge please

**Review by youaresofunny — APPROVED** [review]

**Pull request merged**